sdk/storage/storage-blob/src/BlockBlobClient.ts

```diff
@@ -65,6 +67,40 @@ export class BlockBlobClient extends BlobClient {
    */
   private blockBlobContext: BlockBlob;
 
+  /**
+   * Creates an instance of BlockBlobClient.
+   *
+   * @param {string} connectionString Connection string for an Azure storage account.
+   * @param {string} containerName Container name.
+   * @param {string} blobName Blob name.
+   * @param {NewPipelineOptions} [options] Optional. Options to configure the HTTP pipeline.
+   * @memberof BlockBlobClient
+   */
+  constructor(
+    connectionString: string,
+    containerName: string,
+    blobName: string,
+    options?: NewPipelineOptions
+  );
+  /**
+   * Creates an instance of BlockBlobClient.
+   * This method accepts an encoded URL or non-encoded URL pointing to a block blob.
+   * Encoded URL string will NOT be escaped twice, only special characters in URL path will be escaped.
+   * If a blob name includes ? or %, blob name must be encoded in the URL.
+   *
+   * @param {string} url A URL string pointing to Azure Storage block blob, such as
+   *                     "https://myaccount.blob.core.windows.net/mycontainer/blockblob". You can
+   *                     append a SAS if using AnonymousCredential, such as
+   *                     "https://myaccount.blob.core.windows.net/mycontainer/blockblob?sasString".
+   *                     This method accepts an encoded URL or non-encoded URL pointing to a blob.
+   *                     Encoded URL string will NOT be escaped twice, only special characters in URL path will be escaped.
+   *                     However, if a blob name includes ? or %, blob name must be encoded in the URL.
+   *                     Such as a blob named "my?blob%", the URL should be "https://myaccount.blob.core.windows.net/mycontainer/my%3Fblob%25".
+   * @param {Credential} credential Such as AnonymousCredential, SharedKeyCredential or TokenCredential.
+   * @param {NewPipelineOptions} [options] Optional. Options to configure the HTTP pipeline.
+   * @memberof BlockBlobClient
+   */
+  constructor(url: string, credential: Credential, options?: NewPipelineOptions);
```

Comment:
`credential` should have default value `new AnonymousCredential()`

Reply:
I had it optional for other clients in my prototype, but feels that the inconsistency might be confusing so I removed them.

Reply:
What's the conclusion here ?

Reply:
optional credential is in track 2 proposal from the very beginning, and it's one of the great benefit we can bring to customers using SAS URLs. SAS authentication is very common in storage, making it optional could free them from typing long "new AnonoymousCredential()" and importing the type. Not understand why we drop it in the last minute.

Reply:
/cc @bterlson

Reply:
Talked to many people today, and the outcome is that we should make the credential optional per @XiaoningLiu's point. That leaves us with two options: parsing the connection string, or going back to `fromConnectionString` statics. I am going to float an SDK design guideline next week that proposes `fromConnectionString` as the only way Azure SDK Clients should be constructed from connection strings. I'll report back here when I have closure on this point.